### PR TITLE
Add clang-format file and workflow

### LINKED ISSUE #10: adding .clang_format
We should add a .clang_format style and enforce usage of clang_format as action. So we should decide first on the style and minimum clang-format version. We can use the style we use in VecGeom, with 2 spaces indentation, column limit of 120, replace tab with 2 spaces, .... We also have some scripts allowing to check/apply locally clang-format to all files touched in a branch.

Any opinions on the style/version?

### PR DESCRIPTION
**Pull request by drbenmorgan**
Initial clang-format import using VecGeom's style. Add workflow to check/apply changes on pushes and pull requests.

Closes #10

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# SPDX-FileCopyrightText: 2020 CERN
+# SPDX-License-Identifier: Apache-2.0
+---
+Language:        Cpp
+# BasedOnStyle:  LLVM
+AccessModifierOffset: -2
+AlignAfterOpenBracket: Align
+AlignConsecutiveAssignments: true
+# This would be nice to have but seems to also (mis)align function parameters
```
Comment (graeme-a-stewart) [inline .clang-format:9]: This comment leads me to wonder if an updated clang-format would work better (I'm not 100% sure what the problem aluded to is). Maybe @agheata remembers?

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+AlignEscapedNewlinesLeft: true
+AlignOperands:   true
+AlignTrailingComments: true
+AllowAllParametersOfDeclarationOnNextLine: true
+AllowShortBlocksOnASingleLine: false
+AllowShortCaseLabelsOnASingleLine: false
+AllowShortFunctionsOnASingleLine: Inline
+AllowShortIfStatementsOnASingleLine: true
+AllowShortLoopsOnASingleLine: false
+# This option is "deprecated and is retained for backwards compatibility."
```
Comment (graeme-a-stewart) [inline .clang-format:20]: Remove then...

File: .clang-format [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+SpacesBeforeTrailingComments: 1
+SpacesInAngles:  false
+SpacesInContainerLiterals: true
+SpacesInCStyleCastParentheses: false
+SpacesInParentheses: false
+SpacesInSquareBrackets: false
+Standard:        Cpp11
+TabWidth:        2
+UseTab:          Never
+...
```
Comment (graeme-a-stewart) [inline .clang-format:93]: Does this elipsis serve a purpose?

**Review by graeme-a-stewart — CHANGES_REQUESTED** [review]
Looks grand, just a few comments to quickly discuss before merge

File: .clang-format (hunk 3 above)
Comment (drbenmorgan) [reply .clang-format:93]: This is the YAML document delimeter (that goes with the `---` at the top of the document).

**Comment by drbenmorgan** [comment]
One thing I just noted is that the action won't consider CUDA files per this line in the entry point:

https://github.com/HSF/clang-format-action/blob/802a9bf553bed6e1deea1ae330667424526c4eec/entrypoint.sh#L8

Guess we should put a PR in there to either always consider `.cu` files or an option.

**Review by graeme-a-stewart — APPROVED** [review]

**Comment by graeme-a-stewart** [comment]
> Guess we should put a PR in there to either always consider `.cu` files or an option.

Good point. We should take `.cuh` as well.

**Comment by drbenmorgan** [comment]
Proposed in https://github.com/HSF/clang-format-action/pull/1

**Pull request merged**